pkg/storage/mysql/mysqlstore.go

```diff
@@ -0,0 +1,268 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package mysql
+
+import (
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+
+	// Add as per the documentation - https://github.com/go-sql-driver/mysql
+	_ "github.com/go-sql-driver/mysql"
+
+	"github.com/trustbloc/edge-core/pkg/storage"
+)
+
+// Provider represents an MySQL DB implementation of the storage.Provider interface
+type Provider struct {
+	dbURL string
+	db    *sql.DB
+	dbs   map[string]*sqlDBStore
+	sync.RWMutex
+}
```

Comment:
One other thing we need is the ability to add a prefix for all the new MySQL stores that get made under this provider - take a look at `WithDBPrefix` in `couchdbstore.go` and you should be able to add the same thing here